### DISCUSSION TITLE
How to override async provider ?

### DISCUSSION THREAD

**Discussion by woprandi** [rrousselGit/riverpod#1986] [General]
Since `overrideWithValue` has been removed, how to emulate a `AsyncLoading / AsyncError`  state of a async provider with the new `overrideWith` method ?

**Comment by rrousselGit** [comment]
Create a Future/Stream that completes or emit an error when you want

```dart
final provider = FutureProvider<int>(...);
...

final completer= Completer<int>();
final container = ProviderContainer(overrides: [provider.overrideWith((_) => completer.future)]);

expect(container.read(provider), AsyncLoading<int>());


completer.completeError('err');

await expectLater(
  container.read(provider.future),
  throwsA('err'),
);

```

**Reply by woprandi** [reply]
It's much less easier than before...

I'm trying to migrate this simple test with the new API...

```dart
  test('basic', () async {
    final provider = StreamProvider<int>((_) => Stream.value(0));
    final provider2 = Provider((ref) => ref.watch(provider).value != null);

    var container = ProviderContainer(
      overrides: [provider.overrideWithValue(AsyncLoading())],
    );
    expect(container.read(provider2), false);
    container.updateOverrides([provider.overrideWithValue(AsyncData(0))]);
    expect(container.read(provider2), true);
  });
```

**Reply by rrousselGit** [reply]
As said in the changelog, the removal is only temporary. They will be added back at some point.

**Reply by woprandi** [reply]
Thanks for the answer

**Comment by woprandi** [comment]
Since the test API seems less easy than before with async provider, I'm still using `2.0.0-dev.9`. But it's not possible with Flutter 3.7. Do we have any news about the return of `updateWithValue` ?

**Comment by woprandi** [comment]
@rrousselGit any news ?
